### ISSUE TITLE
publish patchnav-min

### ISSUE THREAD

**Issue by dominictarr** [evbogue/minbay#4]
with a few small changes this code could be published as a `patchnav-*` module

https://github.com/evbogue/minbay/blob/master/index.js#L16-L58

It's actually pretty similar to https://github.com/evbogue/minbay/blob/master/index.js#L16-L58
except that it has hardcoded buttons. If it was a patchnav it could take the buttons from `app.menu` (we could update this api to return images, too)

**Comment by evbogue** [comment]
I did a little test and I realized why I had to fork `patchnav-less`. It is because https://github.com/dominictarr/patchnav-less/blob/master/index.js#L27 doesn't work in the browser.

There's a larger discussion here about whether minbay should use hypertabs or not. I think it'd probably be best for us to support both, similar to how patchless offers three different menus. 

I think if we can figure out how to load the stylesheet into the browserified minbay client I can depend on patchnav-*, and I don't need to publish my own module here.

**Comment by dominictarr** [comment]
okay, so things we need to figure out:

* stylesheet loading
* links

if you follow the `patchnav` pattern correctly you'll be able to switch between tabs or not. I think since you are intending to run in the browser, not having tabs (using the browser's tabs) is a good decision. Actually seeing how you (and others) where doing different layouts is what led me to looking for a standard pattern for navigation. I gotta couple of days of work I gotta do before I can get back to this, but we can def sort this out. These problems are all about things that I just went with the fastest way at the time while prototyping.

**Comment by evbogue** [comment]
maybe we could embed the patchnav-less stylesheet into the js as a hyperscript attribute?

**Comment by evbogue** [comment]
@dominictarr I did a quick test of my link changes in Electron, and adding '#'s to everything seems to work just fine. I imagine there's a simpler way to handle adding a hash to every in-app link? My approach seems to be manually adding them every time there's an in-app link.

I also got work, so I understand that!

**Comment by dominictarr** [comment]
update: I tested the patchnav-less and patchnav-basic and they do actually work in a lite client mode as they are. (although two querks - user ids don't work right and channels don't work right in patchnav-less) I've got some other stuff to catch up on, but will come back to this.

**Comment by dominictarr** [comment]
update: if you use the latest patchnav-less or patchnav-basic then you won't have to change links, and minbase will work with other patchless modules

**Comment by evbogue** [comment]
@dominictarr Awesome, I'll rip out those modules and depend on `patchnav-less`.

**Comment by evbogue** [comment]
Done in 692dd85360cd0b6e492e653c1ce7d342cef9780b

I think this opens up the possibility to depend on more of Patchless. Let me see what else I can depend upon.

**Comment by evbogue** [comment]
@dominictarr I was able to refactor minbay to depend on most of `patchapp-threads` as of 98795996bdc9fddfd2d98888c6c483d738f52052. 

The two exceptions are `post.js` and `message.js`, which have various different opinions about where elements should render on the page.

**Comment by dominictarr** [comment]
what are the differences? there ought to be a way to simply override those parts, but if you tell me about how yours is different we might be able to figure out a smoother way to do that.

**Comment by evbogue** [comment]
@dominictarr There are a couple of stylistic differences I've noticed. 

1. I'm using the old `message-link` and `message-name` modules from Patchbay@6 that grab the old thread and summerize it for the reply
2. I'm using `human-time` instead of `nice-ago` for dates
3. I moved reply to the bottom of the message instead of the top

**Comment by dominictarr** [comment]
Right, your reply is a action - my reply is actually just a link to the thread, so it's the same as your bit that uses `message_link`... also you have both `message.meta` and `message.actions` at the bottom.

Another way you could show meta and actions is in a dropdown menu - I'd expect this would be necessary if you really wanted to declutter the UI... we just don't have so many meta/actions implemented yet that it's been necessary to have a dropdown.

meta differ from actions in that they probably show some information so hiding them inside a menu might not be wanted.

The timestamp could really be a meta, too - but in both our designs it's handled as a special case. In patchless it's under the avatar and in minbay it's over it... actually I tried running minbay and it's just using `patchapp-thread/message`

![2018-01-02-111336_507x402_scrot](https://user-images.githubusercontent.com/259374/34471432-440af44e-efae-11e7-89cd-f3aa9d802b4a.png)

also: message_link is a _hard dependency_, that means if it's not provided it will just error and nothing will work... that fails the goal of easy to add and remove things because you can't remove this with a single line change (you'd have to also change the code in `minbay/message`). But there is more to discuss there about how we arrange things around messages...

---

But the thing I more importantly want to discuss to finish up this issue, is to suggest you taking over patchnav-basic. Instead of a hard coded set of links, you should take those from `app.menu`
https://github.com/dominictarr/patchnav-basic/blob/master/index.js#L52-L54
(this would make it easy for someone to add another view + item in that menu without editing the `minbay/index.js`)

To get the design of minbay with a link to your own profile, you could just have another `patchapp` that used `app.menu` (and possibly `app.view` if your own profile has different info on it)

hmm, and maybe use CSS to get the avatar image in there?

**Comment by evbogue** [comment]
@dominictarr 

>  actually I tried running minbay and it's just using patchapp-thread/message

That's because I decided to try depending on it and see if I could work with that. But I still think I have a few different preferences here for Minbay, so the discussion is still relevent.

> But the thing I more importantly want to discuss to finish up this issue, is to suggest you taking over patchnav-basic.

I agree with you that the nav should be generated from the app.menu items. I'll work on figuring out how to load the avatar image into `patchnav-basic`.
